test/unit/ngsiv2/httpBindings-test.js

```diff
@@ -0,0 +1,738 @@
+            .post('/v2/entities/Second%20UL%20Device/attrs',
+                utils.readExampleFile('./test/unit/ngsiv2/contextRequests/multipleMeasure.json'))
+                .reply(204);
+            });
+
+        it('should end up with a 200OK status code', function(done) {
+            request(getOptions, function(error, response, body) {
+                should.not.exist(error);
+                response.statusCode.should.equal(200);
+                done();
+            });
+        });
+        it('should send a new update context request to the Context Broker with those attributes', function(done) {
+            request(getOptions, function(error, response, body) {
+                contextBrokerMock.done();
+                done();
+            });
+        });
+    });
+
+
+    describe('When a new single measure arrives for a Device, via HTTP POST', function() {
+     var getOptions = {
+         url: 'http://localhost:' + config.http.port + '/iot/d',
+         method: 'POST',
+         qs: {
+             i: 'MQTT_2',
+             k: '1234'
+         },
+         headers: {
+             'Content-type': 'text/plain'
+         },
+         body: 'temperature|23'
+     };
+
+     beforeEach(function() {
+         contextBrokerMock
+         .matchHeader('fiware-service', 'smartGondor')
+         .matchHeader('fiware-servicepath', '/gardens')
+         .post('/v2/entities/Second%20UL%20Device/attrs',
+            utils.readExampleFile('./test/unit/ngsiv2/contextRequests/singleMeasure.json'))
+         .reply(204);
+     });
+
+     it('should end up with a 200OK status code', function(done) {
+         request(getOptions, function(error, response, body) {
+             should.not.exist(error);
+             response.statusCode.should.equal(200);
+             done();
+         });
+     });
+     it('should send a new update context request to the Context Broker with just that attribute', function(done) {
+         request(getOptions, function(error, response, body) {
+             contextBrokerMock.done();
+             done();
+         });
+     });
+ });
+
+    describe('When multiple groups of measures arrive, via HTTP POST', function() {
+        var getOptions = {
+            url: 'http://localhost:' + config.http.port + '/iot/d',
+            method: 'POST',
+            qs: {
+                i: 'MQTT_2',
+                k: '1234'
+            },
+            headers: {
+                'Content-type': 'text/plain'
+            },
+            body: 'temperature|23#humidity|98'
+        };
+
+        beforeEach(function() {
+            contextBrokerMock
+            .matchHeader('fiware-service', 'smartGondor')
+            .matchHeader('fiware-servicepath', '/gardens')
+            .post('/v2/entities/Second%20UL%20Device/attrs',
+                utils.readExampleFile('./test/unit/ngsiv2/contextRequests/singleMeasure.json'))
+            .reply(204);
+
+                contextBrokerMock
```

Comment:
Indent (not only within the contextBrokerMock block, but also better this contextBrokerMock block and the other above)